### DISCUSSION TITLE
Issue with text positioning when rotating objects

### DISCUSSION THREAD

**Discussion by matifanger** [fabricjs/fabric.js#9897] [Q&A]
When rotating a Rect with an angle different from 0, the text elements positioned alongside the rect **move further away from its edges, resulting in inconsistent positioning**

Angle 0:
![chrome_nFDN9rgNIq](https://github.com/fabricjs/fabric.js/assets/39188240/32b4db86-39f9-416d-a7a5-47050ec2de63)

When add some angle:
![chrome_aVoNFWAkyS](https://github.com/fabricjs/fabric.js/assets/39188240/088dca81-4612-4e6c-a43b-ac7488294e06)

Code that i did:
* The param "size" specifies the texts that will be shown

```ts
    addSizesTextToArea(area: fabric.Object, size: { width: string, height: string }) {
      const textWidth = new this.fabric.Text(size.width, {
        fontSize: 12,
        fill: 'black',
      })
      const textHeight = new this.fabric.Text(size.height, {
        fontSize: 12,
        fill: 'black',
      })

      textWidth.angle = area.angle
      textHeight.angle = area.angle

      const clipCenter = area.getCenterPoint()
      const areaCoords = area.getBoundingRect()

      const offset = Math.abs(Math.cos(area.angle! * Math.PI / 180)) * 20

      const textWidthOffsetX = (areaCoords.width / 2 + textWidth.width! / 2) + offset
      const pointWidth = this.fabric.util.rotatePoint(
        new this.fabric.Point(clipCenter.x + textWidthOffsetX, clipCenter.y),
        clipCenter,
        this.fabric.util.degreesToRadians(area.angle!),
      )
      textWidth.setPositionByOrigin(pointWidth, 'center', 'center')

      const textHeightOffsetY = -(areaCoords.height / 2 + textHeight.height! / 2) - offset
      const pointHeight = this.fabric.util.rotatePoint(
        new this.fabric.Point(clipCenter.x, clipCenter.y + textHeightOffsetY),
        clipCenter,
        this.fabric.util.degreesToRadians(area.angle!),
      )
      textHeight.setPositionByOrigin(pointHeight, 'center', 'center')

      textWidth.setCoords()
      textHeight.setCoords()

      this.canvas?.add(textWidth)
      this.canvas?.add(textHeight)
    },
```

**Comment by asturur** [comment]
My suggestion to simplify this is to use:
- rect.getCoords to get the 4 corner points of the rect
- use the middle point between tl and tr to get the center point up, and tr and br to get the vertical middle point.
- Then take the text, set the angle and do       textHeight.setPositionByOrigin(pointHeight, 'left', 'center') for the height and 'center' , 'bottom' for the other one so you avoid doing the rotatePoint code where there is something you are doing wrong eventually.


You can also consider maybe as a stretch to use the control API and try to create non interactive controls that render as text, so that the position is handled for you ?
